### ISSUE TITLE
[Testing] Implement a clean way to get the deployed app URL

### ISSUE THREAD

**Issue by meltsufin** [GoogleCloudPlatform/jetty-runtime#93]

**Comment by jmcc0nn3ll** [comment]
This would allow us to do away with the split IntegrationTest setup, LocalIntegrationTest/RemoteIntegrationTest.  Currently the only fundamental difference between the two is this URI, localhost:<random port> for local and the remote URI combining service id, project id and version.  

Now there will likely be differences in tests soon we are currently able to express now, but a cleaner approach down the road might be to use test scoped annotations instead of the <includes/excludes> in the pom.xml.

tl;dr easy access to the URI means we can re-evaluate some testing structure

**Comment by joakime** [comment]
This seems similar to the enhancement request at GoogleCloudPlatform/app-maven-plugin/issues/86

**Comment by jmcc0nn3ll** [comment]
That's the one I was looking for, thanks @joakime
